include/zephyr/drivers/counter.h

```diff
@@ -113,27 +122,36 @@ typedef void (*counter_alarm_callback_t)(const struct device *dev,
 					 void *user_data);
 
 /** @brief Alarm callback structure.
- *
- * @param callback Callback called on alarm (cannot be NULL).
- * @param ticks Number of ticks that triggers the alarm. It can be relative (to
- *		now) or an absolute value (see @ref COUNTER_ALARM_CFG_ABSOLUTE).
- *		Both, relative and absolute, alarm values can be any value
- *		between zero and the current top value (see @ref counter_get_top_value).
- *		When setting an absolute alarm value close to the current counter
- *		value there is a risk that the counter will have counted past the
- *		given absolute value before the driver manages to activate the alarm.
- *		Therefore a guard period can be defined that lets the driver decide
- *		unambiguously whether it is late or not (see @ref counter_set_guard_period).
- *		If the counter is clock driven then ticks can be converted to
- *		microseconds (see @ref counter_ticks_to_us). Alternatively,
- *		the counter implementation may count asynchronous events.
- * @param user_data User data returned in callback.
- * @param flags	Alarm flags. See @ref COUNTER_ALARM_FLAGS.
  */
 struct counter_alarm_cfg {
+	/**
+	 * @brief Callback called on alarm (cannot be NULL).
```

Comment:
do we need the @brief here? looks weird.

Reply:
fair enough, i went for the implicit `@brief` :)